src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/SQLClause.java

```diff
@@ -99,10 +99,7 @@ private SQLMethodInvokeExpr wrapNestedFunction(SQLExpr expr) {
     String pathFromIdentifier(SQLExpr identifier) {
         String field = Util.extendedToString(identifier);
         int lastDot = field.lastIndexOf(".");
-        if (lastDot == -1) {
-            throw new IllegalStateException("pathFromIdentifier() is being invoked on the wrong field [" + field + "]");
-        }
-        return field.substring(0, lastDot);
+        return lastDot == -1 ? field :field.substring(0, lastDot);
```

Comment:
Mostly question: It looks like the IllegalStateException will no longer be thrown with this change? If so, will returning just the field be as descriptive/have the same desired effect?

Reply:
Done.

Reply:
Yes. It is desired behaviour. for example,
expecting p returned as path in both WHERE p.name = 'A' and WHERE p IS NULL cases in which expr.name = p.name and p separately.